File: app/src/main/java/com/wuguangxin/simple/adapter/StringAdapter.java
package com.wuguangxin.simple.adapter;

import android.content.Context;
import android.view.View;
import android.widget.TextView;

import com.wuguangxin.adapter.BaseRecyclerAdapter;
import com.wuguangxin.adapter.BaseViewHolder;
import com.wuguangxin.simple.R;

import java.util.List;

/**
 * Created by wuguangxin on 2021/6/30.
 */
public class StringAdapter extends BaseRecyclerAdapter<String, StringAdapter.MyViewHolder> {

    public StringAdapter(Context context, List<String> list) {
        super(context, list);
    }

    @Override
    public int getLayoutId() {
        return R.layout.item_simple_layout;
    }

    @Override
    public MyViewHolder createViewHolder(View view, int position) {
        return new MyViewHolder(view);
    }

    @Override
    public void bindViewData(MyViewHolder vewHolder, String s, int position, int type) {
        vewHolder.textView.setText(s);
    }

    public static class MyViewHolder extends BaseViewHolder {
        public TextView textView;

        public MyViewHolder(View view) {
            super(view);
          textView = findView(R.id.textview);
        }
    }

}

File: app/src/main/java/com/wuguangxin/simple/adapter/StringAdapter.kt
package com.wuguangxin.simple.adapter

import com.chad.library.adapter.base.BaseQuickAdapter
import com.chad.library.adapter.base.viewholder.BaseViewHolder
import com.wuguangxin.simple.R

/**
 * Created by wuguangxin on 2021/6/30.
 */
class StringAdapter(list: MutableList<String>?) :
    BaseQuickAdapter<String, BaseViewHolder>(R.layout.item_simple_layout, list) {

    override fun convert(holder: BaseViewHolder, item: String) {
        holder.setText(R.id.textview, item)
    }
}